EngineMath/Vector.cpp

```diff
@@ -8,4 +8,76 @@
 
 namespace EngineMath {
 
+	//Public CTORS
+	Vector::Vector() {
+		data = new T[size];
+		for (int i = 0; i < size; i++)
+			data[i] = 0;
+	}
+
+	Vector::Vector(Ts... args) {
+		data = new T[size];
+		data = {args...};
```

Comment:
I don't think this is how variadic data structures work.

I think the syntax you were going for was:

```
template <class T, class ... Args>
class Vector{
    // " Base case " single argument constructor
    Vector(T arg) {
        elements = arg;
    }

    Vector(T arg, Args ... leftoverArgs) : element(arg), tail(leftoverArgs) {

    }

    T element;
    Vector<T, Args...> tail;
}
```

The construction of variadic data types behaves in a recursive fashion (though it's not actually recursive).

Think of it almost like constructing a linked-list of vector elements at compile time. That's kind of how variadic data structures work.

Every call to Vector(T arg, Args ... leftoverArgs) takes an element out of the parameter pack and stores it in a member variable. This continues until there is only one argument left, in which case the "base case" ctor is called. 

Afterwords, you're left with a set of nested Vector structures.